### PR TITLE
Changes in Pixel 2D template making code cfg for Phase-2 + adding a unit test

### PR DESCRIPTION
**Pull request by tvami**
PR description:

Minor modifications to the Pixel 2D template making code for the Phase-2 detector. After #31622 we are able to make charge reweighting in Phase-2, too, which will need the 2D templates.  Also added a unit test which creates a 2D template file.

PR validation:
The command which is in the unit test was run.

if this PR is a backport please specify the original PR and why you need to backport that PR:
This is not a PR, and there is no PR planned

cc: @mmusich

### CODE REVIEW

**Comment by tvami** [comment]
Please test it together with 
https://github.com/cms-data/CalibTracker-SiPixelESProducers/pull/2

**Comment by mmusich** [comment]
test parameters
* pull_request =  cms-data/CalibTracker-SiPixelESProducers#2

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
let's ty again, hopefully with a new IB.

**Comment by mmusich** [comment]
@cmsbuild, please test

File: CondTools/SiPixel/test/BuildFile.xml [hunk 1]
```diff
@@ -19,3 +19,9 @@
   </bin>
 </environment>
 
+<environment>
+  <bin   file="testPixelDBs.cpp">
```
Comment (mmusich) [inline CondTools/SiPixel/test/BuildFile.xml:23]: @tvami this new test used the same `testPixelDBs.cpp` file as the already existing test `run_0T_BoR3`.
Either split it (creating a new `testPixePhaselDBs.cpp`) or put everything in the same bash executable.

File: CondTools/SiPixel/test/run_Phase2cpeDBTest_v6.sh [hunk 2]
```diff
@@ -1,4 +1,10 @@
-cmsRun SiPixelTemplateDBObjectUploader_Phase2_cfg.py MagField=3.8 Version=6 Append=mc_25x100 Map=../data/phase2_T15_mapping.csv geometry=T15 TemplateFilePath=CondTools/SiPixel/data/25by100_dual_slope_temps
```
Comment (mmusich) [inline CondTools/SiPixel/test/run_Phase2cpeDBTest_v6.sh:1]: @tvami this file is missing the shebang.

File: CondTools/SiPixel/test/run_Phase2cpeDBTest_v6.sh [hunk 3]
```diff
@@ -1,4 +1,10 @@
...
+echo "TESTING Pixel CPE DB codes (for Phase-2) ..."
 
+cmsRun  ${LOCAL_TEST_DIR}/SiPixelTemplateDBObjectUploader_Phase2_cfg.py MagField=3.8 Version=6 Append=mc_25x100 Map= ${LOCAL_TEST_DIR}/../data/phase2_T15_mapping.csv geometry=T15 TemplateFilePath= ${LOCAL_TEST_DIR}/CondTools/SiPixel/data/25by100_dual_slope_temps || die "Failure running SiPixelTemplateDBObjectUploader_cfg.py" $? 
+
+cmsRun  ${LOCAL_TEST_DIR}/SiPixelGenErrorDBObjectUploader_Phase2_cfg.py MagField=3.8 Version=6 Append=mc_25x100 Map= ${LOCAL_TEST_DIR}/../data/phase2_T15_mapping.csv geometry=T15 GenErrFilePath= ${LOCAL_TEST_DIR}/CondTools/SiPixel/data/25by100_dual_slope_temps || die "Failure running SiPixelGenErrorDBObjectUploader_Phase2_cfg.py" $? 
+
+#For 2D templates 
+cmsRun  ${LOCAL_TEST_DIR}/SiPixel2DTemplateDBObjectUploader_Phase2_cfg.py MagField=3.8 Version=7 Append=mc_25x100 Map= ${LOCAL_TEST_DIR}/../data/Phase2_25by100_templates_2020October/phase2_T15_mapping.csv TemplateFilePath= ${LOCAL_TEST_DIR}/CondTools/SiPixel/data/Phase2_25by100_templates_2020October denominator=True || die "Failure running SiPixel2DTemplateDBObjectUploader_Phase2_cfg.py" $?
```
Comment (mmusich) [inline CondTools/SiPixel/test/run_Phase2cpeDBTest_v6.sh:10]: while you are at it, please fix the missing newline.

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please abort

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

File: CondTools/SiPixel/test/createTestDBObjects.sh [hunk 4]
```diff
@@ -11,10 +11,10 @@ echo "TESTING Pixel 1D GenErr DB code ..."
 cmsRun ${LOCAL_TEST_DIR}/SiPixelGenErrorDBObjectUploader_cfg.py MagField=0.0 Fullname=SiPixelGenErrorDBObject_phase1_0T_mc_BoR3_v1_bugfix Map=${LOCAL_TEST_DIR}/../data/phaseI_mapping.csv GenErrFilePath=CalibTracker/SiPixelESProducers/data/SiPixelTemplateDBObject_0T_phase1_BoR3_v1 || die "Failure running SiPixelGenErrorDBObjectUploader_cfg.py" $? 
 
 echo "TESTING Pixel 1D Template DB code for Phase-2 ..."
-cmsRun ${LOCAL_TEST_DIR}/SiPixelTemplateDBObjectUploader_Phase2_cfg.py MagField=3.8 Version=7 Append=mc_25x100 Map=${LOCAL_TEST_DIR}/../data/Phase2_25by100_templates_2020October/phase2_T15_mapping.csv geometry=T15 TemplateFilePath=CondTools/SiPixel/data/Phase2_25by100_templates_2020October || die "Failure running SiPixelTemplateDBObjectUploader_Phase2_cfg.py" $? 
+cmsRun ${LOCAL_TEST_DIR}/SiPixelTemplateDBObjectUploader_Phase2_cfg.py MagField=3.8 Version=7 Append=mc_25x100 Map=$CMSSW_BASE/src/CondTools/SiPixel/data/Phase2_25by100_templates_2020October/phase2_T15_mapping.csv geometry=T15 TemplateFilePath=CondTools/SiPixel/data/Phase2_25by100_templates_2020October || die "Failure running SiPixelTemplateDBObjectUploader_Phase2_cfg.py" $?
```
Comment (mrodozov) [inline CondTools/SiPixel/test/createTestDBObjects.sh:14]: I'm not sure CalibTracker-SiPixelESProducers and CondTools/SiPixel are the same thing. Would you try chaning this to CalibTracker/SiPixelESProducers becase your PR is actually for that package/subpackage

**Comment by tvami** [comment]
Hi @mrodozov , I did your suggestion, thanks! Can you please trigger tests?

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Comment by mrodozov** [comment]
After this PR this test:
SiPixelTemplateDBObjectUploader_Phase2_cfg
started failing for the python3 IBs (PY3 and DEVEL). 
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc820/CMSSW_11_2_PY3_X_2020-11-06-2300/unitTestLogs/CondTools/SiPixel#/18645-18645
There are two things incompatible with Py3  - the print statements and the different csv module interface.
The print is easy, but the cvs statements might need be put in if py2 elseif py3 statements.

**Pull request merged**